src/mahoji/commands/tames.ts

```diff
@@ -835,6 +884,17 @@ async function killCommand(user: KlasaUser, channelID: bigint, str: string) {
 		suffix: 'tame'
 	});
 
+	const deathChance = monster.deathChance ? monster.deathChance({ tame }) : 0;
+	const died = percentChance(deathChance);
+	let fakeDuration: number | undefined = undefined;
+	if (died) {
+		const deathKC = randInt(1, quantity);
```

Comment:
I would definitely change this so a death doesn't wipe your entire trip, only the kills you failed. Otherwise it would be better to do a lower quantity and you're punished for having Patreon / Zak.

```ts
const deathChance = monster.deathChance ? monster.deathChance({ tame }) : 0;
let fakeDuration: number | undefined = undefined;
let deaths = 0;
for (let x = 0; x < quantity; x++) {
	if (percentChance(deathChance)) deaths++;
}
if (deaths) {
	// Fake duration is actually the real duration. note to self
	// Reduce quantity by however many deaths there were.
	fakeDuration = ((quantity - deaths) / quantity) * speed;
}
```


Then obviously, just pass the `quantity - deaths` loot for the quantity of the task, and give partial loot